cli/src/main/java/com/devonfw/tools/ide/tool/mvn/Mvn.java

```diff
@@ -248,11 +246,15 @@ public Path getMavenConfFolder(boolean legacy) {
    * @return the maven arguments (MVN_ARGS).
    */
   public String getMavenArgs() {
-    Path mavenConfFolder = getMavenConfFolder(false);
+    Path mavenConfFolder = this.context.getMavenConfigurationFolder();
     Path mvnSettingsFile = mavenConfFolder.resolve(Mvn.SETTINGS_FILE);
     if (!Files.exists(mvnSettingsFile)) {
       return null;
     }
+    Path settingsSecurityFile = mavenConfFolder.resolve(SETTINGS_SECURITY_FILE);
+    if (!Files.exists(settingsSecurityFile)) {
+      return null;
```

Comment:
This is IMHO incorrect. We have two parameters `-s` and `-Dsettings.security=` that both take a file as argument.
We should make this bullet-proof and set each parameter individually based on the existence of their according files.
Only if both files are absent, we should return null.